src/librustc_trans/back/write.rs

```diff
@@ -1035,20 +1055,26 @@ pub unsafe fn with_llvm_pmb(llmod: ModuleRef,
     // always-inline functions (but don't add lifetime intrinsics), at O1 we
     // inline with lifetime intrinsics, and O2+ we add an inliner with a
     // thresholds copied from clang.
-    match (opt, inline_threshold) {
-        (_, Some(t)) => {
+    match (opt_level, opt_size, inline_threshold) {
+        (_, _, Some(t)) => {
             llvm::LLVMPassManagerBuilderUseInlinerWithThreshold(builder, t as u32);
         }
-        (llvm::CodeGenLevelNone, _) => {
+        (_, llvm::CodeGenOptSizeDefault, _) => {
+            llvm::LLVMPassManagerBuilderUseInlinerWithThreshold(builder, 75);
```

Comment:
It looks like this doesn't quite mirror the standard [LLVM logic](https://github.com/llvm-mirror/llvm/blob/54124b63dc32bce6dcbe86d7c0cd54aaf657f09d/lib/Analysis/InlineCost.cpp#L1441)? 

If O3 is enabled, the threshold seems to be [275](https://github.com/llvm-mirror/llvm/blob/54124b63dc32bce6dcbe86d7c0cd54aaf657f09d/lib/Analysis/InlineCost.cpp#L51), then there's size/min (75/25), and finally there's the [default 225](https://github.com/llvm-mirror/llvm/blob/54124b63dc32bce6dcbe86d7c0cd54aaf657f09d/lib/Analysis/InlineCost.cpp#L53).